serde_derive/src/ser.rs

```diff
@@ -409,6 +419,99 @@ fn serialize_internally_tagged_variant(type_ident: &str,
     }
 }
 
+fn serialize_internally_content_tagged_variant(type_ident: &str,
+                                               variant_ident: &str,
+                                               generics: &syn::Generics,
+                                               ty: syn::Ty,
+                                               variant: &Variant,
+                                               item_attrs: &attr::Item,
+                                               tag: &str,
+                                               content: &str)
+                                               -> Tokens {
+    let type_name = item_attrs.name().serialize_name();
+    let variant_name = variant.attrs.name().serialize_name();
+
+    match variant.style {
+        Style::Unit => {
+            quote!({
+                let mut __struct = try!(_serde::Serializer::serialize_struct(
+                    _serializer, #type_name, 1));
+                try!(_serde::ser::SerializeStruct::serialize_field(
+                    &mut __struct, #tag, #variant_name));
+                _serde::ser::SerializeStruct::end(__struct)
+            })
+        }
+        Style::Newtype => {
+            let field = &variant.fields[0];
+            let mut field_expr = quote!(__simple_value);
+            if let Some(path) = field.attrs.serialize_with() {
+                field_expr = wrap_serialize_with(&ty, generics, field.ty, path, field_expr);
+            }
+
+            quote! {
+                _serde::ser::private::serialize_tagged_newtype(
+                    _serializer,
+                    #type_ident,
+                    #variant_ident,
+                    #tag,
+                    #variant_name,
+                    #field_expr,
+                ),
+            }
+        }
+
+        Style::Tuple => {
+            let where_clause = &generics.where_clause;
+
+            let wrapper_generics = aster::from_generics(generics.clone())
+                .add_lifetime_bound("'__a")
+                .lifetime_name("'__a")
+                .build();
+
+            let wrapper_ty = aster::path()
+                .segment("Inner")
+                .with_generics(wrapper_generics.clone())
+                .build()
+                .build();
+
+            let field_count = variant.fields.len();
+            let field_offset = variant.fields.iter().enumerate().map(
+                |(i, _)| syn::Lit::Int(i as u64, syn::IntTy::Unsuffixed));
+
+            let x = quote!({
+                // Define a tuple struct for the interior along with
+                // the necessary serialization logic.
+                struct Inner #wrapper_generics #where_clause (&'__a #ty);
```

Comment:
I know this is WIP but you're going to want to declare just one of these wrappers for the entire enum like I showed in my code, not one per variant. The generics / PhantomData / etc are going to be impossible to get right otherwise.